### DISCUSSION TITLE
CPU and GPU temp graph bar

### DISCUSSION THREAD

**Discussion by jimaiwr** [mathoudebine/turing-smart-screen-python#435] [General]
Feature request? I miss this feature in the original 3.5" turing software.
Is this hard to code? I'm not too bad at python would give it a go if it's not too complex.

If anyone has pointers or a fork that would be great thank you all.
Glad to have found this alternative. Thanks mathoudebine & contributors.

**Comment by jimaiwr** [comment]
also noticed undefined character appearing with the time. Am I missing something in windows fonts? 
![image](https://github.com/mathoudebine/turing-smart-screen-python/assets/59927023/cd09eaf3-ae27-43c6-86b4-95b85037602e)

**Comment by mathoudebine** [comment]
For the graph part, there is a PR #371 that is currently in review. I will try to integrate it soon, and to extend the graph functionality to all metrics.

The fonts used by themes are included in the project sources so the issue with the square does not come from your Windows.
Can you try to edit the theme and use another font to display the time?
What is your Windows language?

**Comment by c0ldJS** [comment]
I've run into an unexpected error while implementing horizontal bars for temperatures. Wanted to raise it here before opening a bug, in case the feature isn't prime time yet or if I've bungled something.

While using release 3.4.0, asking `GRAPH` to `SHOW: True` for CPU or GPU temperatures will crash the theme preview and the following error is returned:
```
 ... lcd_comm.py", line 460, in DisplayRadialProgressBar
     assert 0 < bar_width <= radius, f'Progress bar linewidth is {bar_width}, must be > 0 and <= radius'
 AssertionError: Progress bar linewidth is X, must be > 0 and <= radius
```

... which calls out radial style graphing even though `RADIAL` had not been used. Using `GRAPH` elsewhere in the theme works as expected and rasters horizontal bars to the preview and on hardware.

**Reply by c0ldJS** [reply]
Just opened issue #497 for this.
